cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/preview/DefaultPreviewManager.java

```diff
@@ -215,6 +217,17 @@ public ApplicationId start(@Name("namespaceId") NamespaceId namespace, AppReques
       throw new IllegalStateException("Preview service is not running. Cannot start preview for " + programId);
     }
 
+
+    Map<String, Object> config = appRequest.getConfig();
+    if (config.get(Constants.Preview.PREVIEW_REQUEST_NUM_OF_RECORDS) instanceof Double) {
+      Double numOfRecords = (Double) config.get(Constants.Preview.PREVIEW_REQUEST_NUM_OF_RECORDS);
+      if (numOfRecords >= Double.parseDouble(previewCConf.get(Constants.Preview.MAX_NUM_OF_RECORDS))) {
+        throw new BadRequestException("Preview record limit exceeded");
+      }
+    } else {
+      throw new BadRequestException("Number of preview records not of type Double");
```

Comment:
What if it's null? We handle a default of 100 records. Please ensure it works with null value and null config.

Reply:
Also don't say "Double" in the error message. It's confusing as we expect int